### ISSUE TITLE
g:syntastic_check_on_wq=0 doesn't work as expected

### ISSUE THREAD

**Issue by potmpark** [scrooloose/syntastic#1482]
Setting `g:syntastic_check_on_wq=0` should have the effect of stop automated checking on file writing
and location list from popping out.
My syntastic related cofiguration in `.vimrc`:
```
let g:syntastic_debug = 29
let g:syntastic_mode_map = {
    \ 'mode': 'passive',
    \ 'active_filetypes': [],
    \ 'passive_filetypes': [] }
let g:syntastic_check_on_wq = 0
let g:syntastic_check_on_open = 0
let g:syntastic_auto_jump = 1
let g:syntastic_auto_loc_list = 1
let g:syntastic_enable_signs = 0
let g:syntastic_stl_format = '[%E{E:%fe #%e}%B{, }%W{W:%fw #%w}]'
nnoremap <silent><C-I> :SyntasticCheck<CR>
```
Expected: Saving files using `:w` should not cause syntastic to do automated check or location list
                 popping whether there are errors or not.
Actual: Before using `<C-I>` to do manually checking , everything worked as expected.
            After using `<C-I>`, if errors were found, subsequently saving files would always having the
           location list being popped out with the same old error last time `<C-I>` was performed,
           no matter
          1. The same error were still in the file.
          2. There were new errors.
          3. There were no syntax errors.
Version:
         *  vim: 7.4
        *  os: Ubuntu 14.10
        *  syntastic: 3.6.0-134 (Vim 704, Linux)

**Comment by lcd047** [comment]
Try setting `always_populate_loc_list` to 1 as per the [recommended setting](https://github.com/scrooloose/syntastic#settings):
```vim
let g:syntastic_always_populate_loc_list = 1
```
Also, `<C-I>` is the same as `<Tab>`, which has a special meaning to Vim.  Bind your macro to some other key combination.

**Comment by potmpark** [comment]
I added:
```
let g:syntastic_always_populate_loc_list = 1
```
in `.vimrc` and rebind `:SyntasticCheck` to `<F2>`, still the same

**Comment by lcd047** [comment]
Can you still reproduce the problem with all other plugins disabled, and if you run `:SyntasticCheck` instead of pressing `<F2>`?

**Comment by potmpark** [comment]
Yes, tested with a clean `.vimrc` with only syntastic configuration in it, `:SyntasticCheck` used
manually.

**Comment by lcd047** [comment]
Please test it with all other plugins disabled.  Move them temporarily to some other directory, or something like that.  Don't rely on a plugin manager to disable them.

If you can still reproduce the problem without other plugins:

* please post the output of `:echo exists('##QuitPre')` and of `:autocmd`
* with the clean configuration (no other plugin etc.) open your test file, run `:SyntasticInfo`, and post the output
* then set `g:syntastic_debug` to 25, run `:SyntasticCheck`, run `:mes`, and post the output
* quit Vim, open your test file again, and run these commands (don't do anything else):
```vim
:set verbosefile=syntastic.log
:set verbose=15
:wq
```
* the above will produce a file `syntastic.log`; please upload the file somewhere, and post the link.

**Comment by potmpark** [comment]
I moved all plugins out of `$HOME/.vim/`, all the files in it:
`_assets autoload CONTRIBUTING.md doc LICENCE plugin README.markdown syntax_checkers`
My `.vimrc`
```
let g:syntastic_always_populate_loc_list = 1
let g:syntastic_mode_map = {
    \ 'mode': 'passive',
    \ 'active_filetypes': [],
    \ 'passive_filetypes': [] }
let g:syntastic_check_on_wq = 0
let g:syntastic_check_on_open = 0
let g:syntastic_auto_jump = 1
let g:syntastic_auto_loc_list = 1
let g:syntastic_enable_signs = 0
let g:syntastic_stl_format = '[%E{E:%fe #%e}%B{, }%W{W:%fw #%w}]'
```
Requested output:
`:echo exists('##QuitPre')` output:
`1`
`:autocmd` output:
https://www.dropbox.com/s/8nn8c8dzbnstjlu/autocmd.txt?dl=0
`:SyntasticInfo` output:
```
Syntastic version: 3.6.0-134 (Vim 704, Linux)
Info for filetype: c
Global mode: passive
Filetype c is passive
The current file will not be checked automatically
Available checkers: gcc make
Currently enabled checker: gcc
```
After adding `let g:syntastic_debug=25` to `.vimrc`, `:SyntasticCheck` output:
```
Messages maintainer: Bram Moolenaar <Bram@vim.org>
"a.c" 11L, 87C
syntastic: 0.033558: autocmd: BufEnter, buffer 1 = 'a.c', &buftype = ''
syntastic: 48.699779: &shell = '/bin/bash', &shellcmdflag = '-c', &shellpipe = '2>&1| tee', &shellquote = '', &shellredir = '>%s 2>&1', &shellslash = 0, &shelltemp = 1, &shellxquote = '', &shellxescape = ''
syntastic: 48.701070: g:syntastic_aggregate_errors = 0
syntastic: 48.701070: g:syntastic_always_populate_loc_list = 1
syntastic: 48.701070: g:syntastic_auto_jump = 1
syntastic: 48.701070: g:syntastic_auto_loc_list = 1
syntastic: 48.701070: g:syntastic_check_on_open = 0
syntastic: 48.701070: g:syntastic_check_on_wq = 0
syntastic: 48.701070: g:syntastic_cursor_columns = 1
syntastic: 48.701070: g:syntastic_debug = 25
syntastic: 48.701070: g:syntastic_echo_current_error = 1
syntastic: 48.701070: g:syntastic_enable_balloons = 1
syntastic: 48.701070: g:syntastic_enable_highlighting = 1
syntastic: 48.701070: g:syntastic_enable_signs = 0
syntastic: 48.701070: g:syntastic_error_symbol = '>>'
syntastic: 48.701070: g:syntastic_exit_checks = 1
syntastic: 48.701070: g:syntastic_filetype_map = {}
syntastic: 48.701070: g:syntastic_full_redraws = 1
syntastic: 48.701070: g:syntastic_id_checkers = 1
syntastic: 48.701070: g:syntastic_ignore_extensions = '\c\v^([gx]?z|lzma|bz2)$'
syntastic: 48.701070: g:syntastic_ignore_files = []
syntastic: 48.701070: g:syntastic_loc_list_height = 10
syntastic: 48.701070: g:syntastic_quiet_messages = {}
syntastic: 48.701070: g:syntastic_reuse_loc_lists = 0
syntastic: 48.701070: g:syntastic_shell = '/bin/bash'
syntastic: 48.701070: g:syntastic_sort_aggregated_errors = 1
syntastic: 48.701070: g:syntastic_stl_format = '[%E{E:%fe #%e}%B{, }%W{W:%fw #%w}]'
syntastic: 48.701070: g:syntastic_style_error_symbol = 'S>'
syntastic: 48.701070: g:syntastic_style_warning_symbol = 'S>'
syntastic: 48.701070: g:syntastic_warning_symbol = '>>'
syntastic: 48.704403: UpdateErrors: default checkers
syntastic: 48.704852: CacheErrors: default checkers
syntastic: 48.705977: g:syntastic_aggregate_errors = 0
syntastic: 48.706332: getcwd() = '/home/potmpark/ltp/playground'
syntastic: 48.712975: CacheErrors: Invoking checker: c/gcc
syntastic: 48.716449: SyntasticMake: called with options: {'postprocess': [], 'errorformat': '%-G%f:%s:,%-G%f:%l: %#error: %#(Each undeclared identifier is reported only%.%#,%-G%f:%l: %#error: %#for each function it appears%.%#,%-GIn file included%.%#,%-G %#from %f:%l\,,%f:%l:%c: %trror: %m,%f:%l:%c: %tarning: %m,%f:%l:%c: %m,%f:%l: %trror: %m,%f:%l: %tarning: %m,%f:%l: %m', 'makeprg': 'gcc -x c -fsyntax-only -std=gnu99 -I. -I.. -Iinclude -Iincludes -I../include -I../includes   a.c'}
error list 1 of 1; 8 errors
syntastic: 48.759867: getLocList: checker c/gcc returned 1
(1 of 3) error: ‘aaa’ undeclared (first use in this function)
syntastic: 48.764446: autocmd: BufEnter, buffer 2 = '', &buftype = ''
syntastic: 48.767607: autocmd: BufEnter, buffer 1 = 'a.c', &buftype = ''
‘aaa’ undeclared (first use in this function)
```
`syntastic.log`
https://www.dropbox.com/s/bcd633p20i0gmzo/syntastic.log?dl=0
The test file:
https://www.dropbox.com/s/2wrv5nf9ukfk5mz/test.c?dl=0

**Comment by lcd047** [comment]
> `"if g:syntastic_check_on_wq`

You're running modified sources. _shrug_

**Comment by lcd047** [comment]
Ok, I think I figured it out what you're saying: if you run the checker, then close the error window, then write the file, the error window would pop back up.  This is now fixed in d26c7f7.

Your report has been quite misleading.  What was going on was not that the checker was run again (it wasn't), but the write triggered a notification refresh without the loclist being refreshed, so the old errors would pop up again.  The problem also had nothing to do with `check_on_wq`, but I can now understand why you were lead to believe it did.  If you didn't have signs turned off you would have noticed that closing the error window doesn't make syntastic "go away".  To reset syntastic you have to run `:SyntasticReset`, not `lclose` (or `quit` in the error window).  Anyway, there was indeed a bug, and it's now (hopefully) fixed.

**Comment by potmpark** [comment]
Works fine now, thanks for the effort.
